### ISSUE TITLE
Allow to get unified legend when using split.by = "seurat_clusters" without having to use split_seurat = TRUE in DimPlot_scCustom()

### ISSUE THREAD

**Issue by CroixJeremy2** [samuel-marsh/scCustomize#171]
Hello,
It's me again and my silly questions. I have a request when using split.by = "seurat_clusters" in DimPlot_scCustom(). Could that be possible to add an option in DimPlot_scCustom() that would allow the user to collect/unify the legend similar to what is plotted when using split_seurat = TRUE ? In other words, I would like to have the display of split_seurat = TRUE for the graphs, but I would like to have the display of split_seurat = FALSE for the legend. Could that be possible to add such option please?

Reproducible example:

```r
library(Seurat)
library(SeuratData)
library(scCustomize)

data("pbmc3k")
pbmc3k = UpdateSeuratObject(pbmc3k)

pbmc3k = NormalizeData(object = pbmc3k)
pbmc3k = FindVariableFeatures(object = pbmc3k)
pbmc3k = ScaleData(object = pbmc3k)
pbmc3k = RunPCA(object = pbmc3k)
pbmc3k = RunUMAP(object = pbmc3k, dims = 1:20)
pbmc3k = FindNeighbors(object = pbmc3k, dims = 1:20)
pbmc3k = FindClusters(object = pbmc3k, resolution = 0.50)
```

Graph part that I would like:
```r
DimPlot_scCustom(seurat_object = pbmc3k, group.by = "seurat_clusters", split.by = "seurat_clusters", num_columns = 3)
```
![image1](https://github.com/samuel-marsh/scCustomize/assets/27291483/cdbf927d-17d5-4bd1-ab6c-8174957376fd)

Legend part that I would like:
```r
DimPlot_scCustom(seurat_object = pbmc3k, group.by = "seurat_clusters", split.by = "seurat_clusters", num_columns = 3, split_seurat = TRUE)
```
![image2](https://github.com/samuel-marsh/scCustomize/assets/27291483/06743399-1345-4be3-8dcf-b30ad9ec1784)

**Additional remark:**
I don't know if this is intentional, but when the number of idents (or groups in group.by) is not divisible by num_columns (in terms of without a "leftover" in the euclidian division), then, the x axis can look weird/shifted as we can see in the second image. Is is possible to force it to be at the bottom as in the first two columns we see on the second image?

Thanks in advance for your response, and for all your work, (and sorry if this feature already exists in case I missed it)
Best regards,

<details>

```r
sessionInfo()
```

R version 4.3.1 (2023-06-16)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Monterey 12.7.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] C

time zone: Europe/Paris
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] scCustomize_2.1.2.9022  pbmc3k.SeuratData_3.1.4 ifnb.SeuratData_3.1.0  
[4] SeuratData_0.2.2.9001   Seurat_5.0.3            SeuratObject_5.0.1     
[7] sp_2.1-3               

loaded via a namespace (and not attached):
  [1] RColorBrewer_1.1-3     jsonlite_1.8.8         shape_1.4.6.1         
  [4] magrittr_2.0.3         spatstat.utils_3.0-4   ggbeeswarm_0.7.2      
  [7] farver_2.1.1           GlobalOptions_0.1.2    vctrs_0.6.5           
 [10] ROCR_1.0-11            spatstat.explore_3.2-6 paletteer_1.6.0       
 [13] janitor_2.2.0          htmltools_0.5.7        forcats_1.0.0         
 [16] sctransform_0.4.1      parallelly_1.37.1      KernSmooth_2.23-22    
 [19] htmlwidgets_1.6.4      ica_1.0-3              plyr_1.8.9            
 [22] lubridate_1.9.3        plotly_4.10.4          zoo_1.8-12            
 [25] igraph_2.0.3           mime_0.12              lifecycle_1.0.4       
 [28] pkgconfig_2.0.3        Matrix_1.6-5           R6_2.5.1              
 [31] fastmap_1.1.1          snakecase_0.11.1       fitdistrplus_1.1-11   
 [34] future_1.33.1          shiny_1.8.0            digest_0.6.35         
 [37] colorspace_2.1-0       rematch2_2.1.2         patchwork_1.2.0       
 [40] tensor_1.5             prismatic_1.1.1        RSpectra_0.16-1       
 [43] irlba_2.3.5.1          labeling_0.4.3         progressr_0.14.0      
 [46] timechange_0.3.0       fansi_1.0.6            spatstat.sparse_3.0-3 
 [49] httr_1.4.7             polyclip_1.10-6        abind_1.4-5           
 [52] compiler_4.3.1         withr_3.0.0            fastDummies_1.7.3     
 [55] MASS_7.3-60.0.1        rappdirs_0.3.3         tools_4.3.1           
 [58] vipor_0.4.7            lmtest_0.9-40          beeswarm_0.4.0        
 [61] httpuv_1.6.14          future.apply_1.11.1    goftest_1.2-3         
 [64] glue_1.7.0             nlme_3.1-164           promises_1.2.1        
 [67] grid_4.3.1             Rtsne_0.17             cluster_2.1.6         
 [70] reshape2_1.4.4         generics_0.1.3         gtable_0.3.4          
 [73] spatstat.data_3.0-4    tidyr_1.3.1            data.table_1.15.2     
 [76] utf8_1.2.4             spatstat.geom_3.2-9    RcppAnnoy_0.0.22      
 [79] ggrepel_0.9.5          RANN_2.6.1             pillar_1.9.0          
 [82] stringr_1.5.1          spam_2.10-0            RcppHNSW_0.6.0        
 [85] ggprism_1.0.4          later_1.3.2            circlize_0.4.16       
 [88] splines_4.3.1          dplyr_1.1.4            lattice_0.22-5        
 [91] survival_3.5-7         deldir_2.0-4           tidyselect_1.2.1      
 [94] miniUI_0.1.1.1         pbapply_1.7-2          gridExtra_2.3         
 [97] scattermore_1.2        matrixStats_1.2.0      stringi_1.8.3         
[100] lazyeval_0.2.2         codetools_0.2-19       tibble_3.2.1          
[103] cli_3.6.2              uwot_0.1.16            xtable_1.8-4          
[106] reticulate_1.35.0      munsell_0.5.0          Rcpp_1.0.12           
[109] globals_0.16.3         spatstat.random_3.2-3  png_0.1-8             
[112] ggrastr_1.0.2          parallel_4.3.1         ellipsis_0.3.2        
[115] ggplot2_3.5.0          dotCall64_1.1-1        listenv_0.9.1         
[118] viridisLite_0.4.2      scales_1.3.0           ggridges_0.5.6        
[121] leiden_0.4.3.1         purrr_1.0.2            crayon_1.5.2          
[124] rlang_1.1.3            cowplot_1.1.3

**Comment by samuel-marsh** [comment]
Hi @CroixJeremy2,

No worries, happy to help.  Thanks for reporting the first issue.  That is new behavior with ggplot2 update to 3.5.0 (I just checked and works fine with ggplot2 3.4.4) so i will work on fix for that to unify the legends because that should be the normal behavior.

In regards to the additional comment, unfortunately nothing I can do there as that is the way Seurat formats the plots (and part of the reason that I changed default behavior in `DimPlot_scCustom`.

Best,
Sam

**Comment by samuel-marsh** [comment]
Hi @CroixJeremy2,

Until I can decide exactly how I want to proceed with this going forward I have added a temporary (may become permanent) fix for users with ggplot2 >= 3.5.0.  The fix is live in develop branch > v2.1.2.9029.  It works for me in testing.

If you still have issues with this after updating to the develop branch please post here and I will reopen the issue.

Best,
Sam
